### PR TITLE
[Java] Fix compilation error for a Map<String, Set<InnerEnum>> due to incompatible types

### LINKED ISSUE #19393: [BUG][JAVA] Compilation error for a `Map<String, Set<InnerEnum>>` due to incompatible types
##### Description

I have the following schema that describes the different roles an employee can have in different projects as a map of collection of enum, e.g.:

```yaml
    Employee:
      type: object
      properties:
        projectRoles:
          type: object
          additionalProperties:
            uniqueItems: true
            type: array
            items:
              type: string
              enum:
                - DEVELOPER
                - TESTER
                - OWNER
```

This works correctly with most generators, however the `java` generator seems to have some trouble with it and generates an invalid `Employee.java class`:

```java
public class Employee {

  // ... truncated

  // This should be Map<String, Set<InnerEnum>> instead
  private Map<String, Set<String>> projectRoles = new HashMap<>();

  // ... truncated

  public Employee putProjectRolesItem(String key, Set<InnerEnum> projectRolesItem) {
    if (this.projectRoles == null) {
      this.projectRoles = new HashMap<>();
    }
    this.projectRoles.put(key, projectRolesItem); // Compiler error here due to the mismatching types
    return this;
  }
}
```

The compiler error is as follows

```
incompatible types: java.util.Set<org.openapitools.client.model.Employee.InnerEnum> cannot be converted to java.util.Set<java.lang.String>
```

Note that the issue only occurs for inline enums (i.e. `InnerEnum`s); enum references work as expected.

##### openapi-generator version

Tested using `openapi-generator` version 7.8.0.

##### OpenAPI declaration file content

A gist with a minimal OpenAPI spec file to reproduce the issue can be found [here](https://gist.github.com/Bragolgirith/5161e67882e5c4b39b8b07b3990e38cd).

##### Generation Details

I'm generating the Java client using the `openapi-generator-maven-plugin` as follows:

```xml
<plugin>
    <groupId>org.openapitools</groupId>
    <artifactId>openapi-generator-maven-plugin</artifactId>
    <version>7.8.0</version>
    <executions>
        <execution>
            <phase>generate-sources</phase>
            <goals>
                <goal>generate</goal>
            </goals>
            <configuration>
                <inputSpec>
                    ${project.basedir}/src/main/resources/employee-api.json
                </inputSpec>
                <generatorName>java</generatorName>
                <library>webclient</library>
                <generateModels>true</generateModels>
                <generateModelTests>false</generateModelTests>
                <generateApiTests>false</generateApiTests>
                <configOptions>
                    <useJakartaEe>true</useJakartaEe>
                </configOptions>
            </configuration>
        </execution>
    </executions>
</plugin>
```

##### Steps to reproduce

1. Generate a Java client using the provided OpenAPI spec file
2. Try to compile the generated client

### PR DESCRIPTION
**Pull request by Bragolgirith**
Fixes #19393.

### CODE REVIEW

File: modules/openapi-generator/src/test/resources/3_0/issue_19393_map_of_inner_enum.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          type: object
+          additionalProperties:
+            uniqueItems: true
+            type: array
+            items:
+              type: string
+              enum:
+                - DEVELOPER
+                - TESTER
+                - OWNER
```
Comment (wing328) [inline modules/openapi-generator/src/test/resources/3_0/issue_19393_map_of_inner_enum.yaml:51]: i did a test with this spec but still got errors:
```
[ERROR] C:\Users\wing3\AppData\Local\Temp\java-enum\src\main\java\org\openapitools\client\model\Employee.java:[118,9] error: enum InnerEnum is already defined in class Employee
[ERROR] C:\Users\wing3\AppData\Local\Temp\java-enum\src\main\java\org\openapitools\client\JSON.java:[49,7] error: duplicate class: org.openapitools.client.JSON

```

can you please take a look?

command: 
```
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g java -i modules/openapi-generator/src/test/resources/3_0/issue_19393_map_of_inner_enum.yaml -o /tmp/java-enum/
```

Comment (Bragolgirith) [reply modules/openapi-generator/src/test/resources/3_0/issue_19393_map_of_inner_enum.yaml:40]: Oh, sorry about that! 

The errors seem to be due to a _different_ bug where the same name (`InnerEnum`) gets generated for multiple enums within the same schema. 

I've now updated the test spec to use two different schemas.

**Comment by wing328** [comment]
@Bragolgirith can you please resolve the merge conflicts when you've time?

cc 
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @martin-mfg (2023/08)

**Review by martin-mfg — APPROVED** [review]

**Comment by wing328** [comment]
FYI. Tested again with the test spec and the output compiles without issue

**Pull request merged**